### ISSUE TITLE
packetfence-perl: Debian build should be done differently

### ISSUE THREAD

**Issue by nqb** [inverse-inc/packetfence#7033]
**Describe the bug**
Currently, if we compared Perl modules provided by packetfence-perl RPM and Debian packages, there is a lot of difference.

There should not have any difference because package build is using same source of truth: dependencies.csv file.

However, it looks like the way we build packetfence-perl on Debian host is not correct because some existing Perl modules installed through packaging are considered as already available for `cpan`.

We must ensure that Debian build are done in a way where existing Perl modules are ignored.

A possible solution is to perform a build on a bare Debian host.

Builds of packetfence-perl 1.2.X have been done on Docker images used to build packages : they certainly contain non necessary Perl modules. We should try with Debian minimal images.

**Comment by nqb** [comment]
Moving milestone.

**Comment by ianbmacdonald** [comment]
I just stumbled on this looking for Debian related issues prior to opening another bug.    I will just say, one of the reasons I selected packetfence years ago, was because it operated on a Debian system, leveraging the package-maintained, secure, components that adhere to Debian policy. 

I would have thought backported CVEs in Debian Perl is just one less thing developers need to worry about, rather than having to unit test against upstream modules that might have new functionality, every time a CVE is published, which can be difficult to stay on top of. 

https://www.opencve.io/cve?vendor=perl

Perl, like many other Debian components, benefits from fast CVE patching and stability over some of the RPM alternatives,  and cpan upstream sources, that typically move upstream when patching CVEs, often with breaking functionality that is avoided only by pinned versions from CPAN, often left vulnerable as developers focus on the latest versions. Perl in particular has a lot of activity in this domain, wrt to security, so it is one area where I really appreciate the intermediate package maintainer role that Debian policy provides. 

While I understand the benefit of bringing the build together from a more static starting point,  will this type of change diminish the security posture of running packetfence on Debian vs other less security focused platforms?

What I am really wondering I suppose, is will this expose any scripts run in the public packetfence portal listeners, or is perl use limited to backend operations where CVE vectors are limited to local access use cases?
